polyfill/lib/calendar.mjs

```diff
@@ -204,9 +286,25 @@ impl['iso8601'] = {
     return GetSlot(date, ISO_DAY);
   },
   era(date) {
-    if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.Date%'));
+    if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
     return undefined;
   },
+  eraYear(date) {
+    if (!HasSlot(date, ISO_YEAR)) date = ES.ToTemporalDate(date, GetIntrinsic('%Temporal.PlainDate%'));
+    return GetSlot(date, ISO_YEAR);
```

Comment:
I think that it'd be better to have valid values for the ISO calendar, because it'd make it easier to write cross-calendar code. Otherwise, writing cross-calendar libraries would require a lot more special casing and would introduce more bugs. This is especially true for `monthCode` which is required for PlainMonthDay in lunisolar calendars.

Here's what I had in mind: 
* `monthCode` - always `${month}`
* `monthType` - always `'regular'`
* `eraYear` - always `year`
* `era` - `undefined` (just like era-less calendars like Hebrew and Chinese)

What do you think?